numpy/core/arrayprint.py

```diff
@@ -781,6 +789,16 @@ def __call__(self, x):
             return self.format % x.astype('i8')
 
 
+class SubArrayFormat(object):
+    def __init__(self, format_function):
+        self.format_function = format_function
+
+    def __call__(self, arr):
+        if arr.ndim <= 1:
+            return "[" + ", ".join(self.format_function(a) for a in arr) + "]"
+        return "[" + ", ".join(self.__call__(a) for a in arr) + "]"
```

Comment:
I think so too. First, I tried it a little and found it's not easy. So, I made `SubArrayFormat` class as the second way.